### PR TITLE
hw26.1

### PR DESCRIPTION
**Pull request by matviichuk92**
Наступні рази буду додавати завдання логічними частинами. 

implemented registration users and delete them
able to add goods and add them to shopping cart

### CODE REVIEW

File: src/main/java/online/shop/controller/DeleteUserController.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("online.shop");
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String userId = req.getParameter("id");
+        Long id = Long.valueOf(userId);
+        userService.delete(id);
```
Comment (AnaLanda) [inline src/main/java/online/shop/controller/DeleteUserController.java:23]: I've been advised by our mentors to delete the user's shopping cart when we delete the user :) So maybe you should add the deletion of the shopping cart here)

File: src/main/webapp/web.xml [hunk 2]
```diff
@@ -3,12 +3,85 @@
...
+        <servlet-class>online.shop.controller.RegistrationController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>registration</servlet-name>
+        <url-pattern>/registration</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>add</servlet-name>
+        <servlet-class>online.shop.controller.ProductAddController</servlet-class>
```
Comment (AnaLanda) [inline src/main/webapp/web.xml]:
```suggestion
        <servlet-class>online.shop.controller.AddProductController</servlet-class>
```

File: src/main/webapp/web.xml [hunk 3]
```diff
@@ -3,12 +3,85 @@
...
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>add</servlet-name>
+        <url-pattern>/add</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>displayShoppingCart</servlet-name>
+        <servlet-class>online.shop.controller.ShoppingCartDisplayController</servlet-class>
```
Comment (AnaLanda) [inline src/main/webapp/web.xml]:
```suggestion
        <servlet-class>online.shop.controller.DisplayShoppingCartController</servlet-class>
```

File: src/main/webapp/web.xml [hunk 4]
```diff
@@ -3,12 +3,85 @@
...
+        <servlet-class>online.shop.controller.UsersController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>allUsers</servlet-name>
+        <url-pattern>/users/all</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allProduct</servlet-name>
+        <servlet-class>online.shop.controller.ProductGetAllController</servlet-class>
```
Comment (AnaLanda) [inline src/main/webapp/web.xml]:
```suggestion
        <servlet-class>online.shop.controller.GetAllProductsController</servlet-class>
```

**Review by AnaLanda — COMMENTED** [review]
Looks great! I added only a couple of minor suggestions)

File: src/main/java/online/shop/controller/DeleteUserController.java (hunk 1 above)
Comment (RomanDubovskyi) [reply src/main/java/online/shop/controller/DeleteUserController.java:23]: \+

File: src/main/java/online/shop/controller/InjectDataController.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import online.shop.lib.Injector;
+import online.shop.service.UserService;
+
+public class InjectDataController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/online/shop/controller/InjectDataController.java:16]: i don't see injecting any data in this controller, what is its purpose ?

File: src/main/java/online/shop/controller/ProductAddController.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import online.shop.lib.Injector;
+import online.shop.model.Product;
+import online.shop.service.ProductService;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class ProductAddController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/online/shop/controller/ProductAddController.java]:
```suggestion
public class AddProductController extends HttpServlet {
```

File: src/main/java/online/shop/controller/ProductGetAllController.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import online.shop.model.Product;
+import online.shop.service.ProductService;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+
+public class ProductGetAllController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/online/shop/controller/ProductGetAllController.java]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

File: src/main/java/online/shop/controller/ShoppingCartDisplayController.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import online.shop.lib.Injector;
+import online.shop.model.Product;
+import online.shop.service.ShoppingCartService;
+import javax.servlet.ServletException;
+
+public class ShoppingCartDisplayController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/online/shop/controller/ShoppingCartDisplayController.java]:
```suggestion
public class DisplayShoppingCartController extends HttpServlet {
```

File: src/main/java/online/shop/controller/UsersController.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import online.shop.lib.Injector;
+import online.shop.model.User;
+import online.shop.service.UserService;
+
+public class UsersController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/online/shop/controller/UsersController.java]:
```suggestion
public class GetAllUsersController extends HttpServlet {
```

File: src/main/webapp/web.xml [hunk 10]
```diff
@@ -3,12 +3,85 @@
...
+        <url-pattern>/shopping-cart</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addProductToCart</servlet-name>
+        <servlet-class>online.shop.controller.AddProductToShoppingCartController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>addProductToCart</servlet-name>
+        <url-pattern>/shopping-cart/add</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]: fix all your url patterns according to checklist example 
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/online/shop/controller/DeleteUserController.java [hunk 11]
```diff
@@ -11,13 +12,16 @@
...
+    private final ShoppingCartService shoppingCartService = (ShoppingCartService) injector
+            .getInstance(ShoppingCartService.class);
 
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
         String userId = req.getParameter("id");
         Long id = Long.valueOf(userId);
         userService.delete(id);
+        shoppingCartService.delete(id);
```
Comment (RomanDubovskyi) [inline src/main/java/online/shop/controller/DeleteUserController.java]: user shopping cart id not user id to delete shopping cart

File: src/main/webapp/web.xml [hunk 12]
```diff
@@ -3,12 +3,85 @@
...
+        <url-pattern>/users/all</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allProduct</servlet-name>
+        <servlet-class>online.shop.controller.GetAllProductsController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>allProduct</servlet-name>
+        <url-pattern>/product/get-all</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/product/all</url-pattern>
```

**Review by RomanDubovskyi — APPROVED** [review]
fix cart deleting before merging

**Pull request merged**